modules/gapi/src/backends/cpu/gcpuimgproc.cpp

```diff
@@ -335,6 +335,17 @@ GAPI_OCV_KERNEL(GCPUNV12toRGBp, cv::gapi::imgproc::GNV12toRGBp)
     }
 };
 
+GAPI_OCV_KERNEL(GCPUNV12toGray, cv::gapi::imgproc::GNV12toGray)
+{
+    static void run(const cv::Mat& in_y, const cv::Mat& in_uv, cv::Mat& out)
+    {
+        cv::Mat yuv_planar;
+        cv::Mat uv_planar(in_y.rows / 2, in_y.cols, CV_8UC1, in_uv.data);
+        cv::vconcat(in_y, uv_planar, yuv_planar);
+        cv::cvtColor(yuv_planar, out, cv::COLOR_YUV2GRAY_NV12);
```

Comment:
OpenCV has got many conversion codes from various YUV formats to gray but it's just alias to `COLOR_YUV2GRAY_420`:

```
  cv::COLOR_YUV2GRAY_NV21 = COLOR_YUV2GRAY_420,
  cv::COLOR_YUV2GRAY_NV12 = COLOR_YUV2GRAY_420,
  cv::COLOR_YUV2GRAY_YV12 = COLOR_YUV2GRAY_420,
  cv::COLOR_YUV2GRAY_IYUV = COLOR_YUV2GRAY_420,
  cv::COLOR_YUV2GRAY_I420 = COLOR_YUV2GRAY_420,
  cv::COLOR_YUV420sp2GRAY = COLOR_YUV2GRAY_420,
  cv::COLOR_YUV420p2GRAY  = COLOR_YUV2GRAY_420
```

`cvtColor` function requires a planar YUV format for this conversion

To create a planar Mat we simply concatenate 2 planes vertically using the `cv::vconcat` function

But we can 't just concatenate 2 planes, because the UV mat is stored in `CV_8U2` format and has a width half that of Y mat. So just create a wrapper with correct format and sizes for `cv::vconcat` function


Anyway, we copy data when concatenating planes, I think it makes sense to have an analog of this function that accepts planar YUV as input. 

@dmatveev What do you think ? Is it requires in this PR ?

Reply:
It is not required in this PR I believe. Thanks,D

Reply:
However, in order to better manage this temporary memory, I believe it can become a compound kernel.

Reply:
Done